js/components/Apple2.tsx

```diff
@@ -53,26 +55,30 @@ export const Apple2 = (props: Apple2Props) => {
                 ...props,
             };
             const apple2 = new Apple2Impl(options);
-            apple2.ready.then(() => {
-                setApple2(apple2);
-                const io = apple2.getIO();
-                const cpu = apple2.getCPU();
-                setIO(io);
-                setCPU(cpu);
-                apple2.reset();
-                apple2.run();
-            }).catch((e) => setError(e));
+            noAwait((async () => {
+                try {
+                    await apple2.ready;
+                    setApple2(apple2);
+                    setIO(apple2.getIO());
+                    setCPU(apple2.getCPU());
+                    await drivesReady.ready;
+                    apple2.reset();
+                    apple2.run();
+                } catch (e) {
+                    setError(e);
+                }
+            }))();
```

Comment:
I propose that (in a different change) we split `noAwait` into two different parts: one that just changes the function's return type to `void` and another one that also invokes the function. Something like this:

```tsx
noAwait(async () => { ... }); // types the function to return void AND invokes the function

<Modal onChange={typeNoAwait(myAsyncCallback)} /> // just types the function
```

What do you think? (Name suggestions are also welcome.)

Reply:
Maybe something like `asyncBlock` and `asyncCallback`?